mayastor-test/e2e/basic_volume_io/basic_volume_io_test.go

```diff
@@ -0,0 +1,73 @@
+// JIRA: CAS-505
+// JIRA: CAS-506
+package basic_volume_io_test
+
+import (
+	"e2e-basic/common"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var defTimeoutSecs = "90s"
+
+func TestBasicVolumeIO(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Basic volume IO tests, NVMe-oF TCP and iSCSI")
+}
+
+func basicVolumeIOTest(scName string) {
+	volName := "basic-vol-io-test-" + scName
+	// Create the volume
+	common.MkPVC(volName, scName)
+
+	// Create the fio Pod
+	fioPodName := "fio-" + volName
+	pod, err := common.CreateFioPod(fioPodName, volName)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(pod).ToNot(BeNil())
+
+	// Wait for the fio Pod to transition to running
+	Eventually(func() bool {
+		return common.IsPodRunning(fioPodName)
+	},
+		defTimeoutSecs,
+		"1s",
+	).Should(Equal(true))
+
+	// Run the fio test
+	common.RunFio(fioPodName, 20)
+
+	// Delete the fio pod
+	err = common.DeletePod(fioPodName)
+	Expect(err).ToNot(HaveOccurred())
+
+	// Delete the volume
+	common.RmPVC(volName, scName)
```

Comment:
If anything in the test fails, the PVC will be leaked. Can we clean up PVCs in after-hook instead? And IMHO we should wait until PV, PVC, MSV really disappears before continuing (could be a utility function in common module).